pkg/handlers/generic/mutation/extraapiservercertsans/inject.go

```diff
@@ -83,15 +100,27 @@ func (h *extraAPIServerCertSANsPatchHandler) Mutate(
 		func(obj *controlplanev1.KubeadmControlPlaneTemplate) error {
 			log.WithValues(
 				"patchedObjectKind", obj.GetObjectKind().GroupVersionKind().String(),
-				"patchedObjectName", client.ObjectKeyFromObject(obj),
+				"patchedObjectName", ctrlclient.ObjectKeyFromObject(obj),
 			).Info("adding API server extra cert SANs in kubeadm config spec")
 
 			if obj.Spec.Template.Spec.KubeadmConfigSpec.ClusterConfiguration == nil {
 				obj.Spec.Template.Spec.KubeadmConfigSpec.ClusterConfiguration = &bootstrapv1.ClusterConfiguration{}
 			}
-			obj.Spec.Template.Spec.KubeadmConfigSpec.ClusterConfiguration.APIServer.CertSANs = extraAPIServerCertSANsVar
-
+			obj.Spec.Template.Spec.KubeadmConfigSpec.ClusterConfiguration.APIServer.CertSANs = extraSans
 			return nil
 		},
 	)
 }
+
+func getDefaultAPIServerSANs(cluster *capiv1.Cluster) []string {
```

Comment:
We now also have it in the Nutanix CC, please handle that one too

https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/15224874e594e057403b7d521aabd40a44dc5053/charts/cluster-api-runtime-extensions-nutanix/defaultclusterclasses/nutanix-cluster-class.yaml#L124-L128

Reply:
I'll stack to this implementation.